sdk/src/test/java/com/mapbox/maps/renderer/MapboxRenderThreadTest.kt

```diff
@@ -471,4 +471,56 @@ class MapboxRenderThreadTest {
     // EGL should be prepared
     assert(mapboxRenderThread.eglPrepared)
   }
+
+  @Test
+  fun snapshotIsTakenAfterDrawAndBeforeSwapBuffers() {
+    val latch = CountDownLatch(1)
+    val surface = mockk<Surface>()
+    every { surface.isValid } returns true
+    every { eglCore.eglStatusSuccess } returns true
+    every { eglCore.createWindowSurface(any()) } returns mockk(relaxed = true)
+    mapboxRenderThread.onSurfaceCreated(surface, 1, 1)
+    val runnable = mockk<Runnable>(relaxUnitFun = true)
+    every { runnable.run() } answers { latch.countDown() }
+    Shadows.shadowOf(renderHandlerThread.handler?.looper).pause()
+    mapboxRenderThread.queueSnapshot(runnable)
+    Shadows.shadowOf(renderHandlerThread.handler?.looper).idle()
+    if (!latch.await(waitTime, TimeUnit.MILLISECONDS)) {
+      throw TimeoutException()
+    }
+    verifyOrder {
+      mapboxRenderer.onDrawFrame()
+      runnable.run()
+      eglCore.swapBuffers(any())
+    }
+  }
+
+  @Test
+  fun multipleSnapshotsAreTaken() {
+    val latch = CountDownLatch(3)
+    val surface = mockk<Surface>()
+    every { surface.isValid } returns true
+    every { eglCore.eglStatusSuccess } returns true
+    every { eglCore.createWindowSurface(any()) } returns mockk(relaxed = true)
+    mapboxRenderThread.onSurfaceCreated(surface, 1, 1)
+    val runnable = mockk<Runnable>(relaxUnitFun = true)
+    val runnable2 = mockk<Runnable>(relaxUnitFun = true)
+    val runnable3 = mockk<Runnable>(relaxUnitFun = true)
+    every { runnable.run() } answers { latch.countDown() }
+    every { runnable2.run() } answers { latch.countDown() }
+    every { runnable3.run() } answers { latch.countDown() }
+    Shadows.shadowOf(renderHandlerThread.handler?.looper).pause()
+    mapboxRenderThread.queueSnapshot(runnable)
+    mapboxRenderThread.queueSnapshot(runnable2)
+    mapboxRenderThread.queueSnapshot(runnable3)
+    Shadows.shadowOf(renderHandlerThread.handler?.looper).idle()
+    if (!latch.await(waitTime, TimeUnit.MILLISECONDS)) {
+      throw TimeoutException()
+    }
+    verify {
```

Comment:
Guess we should also verify order here to make sure that 3 runnalbes are run in correct order and between `mapboxRenderer.onDrawFrame()` and `eglCore.swapBuffers(any())`. Perhaps we could even have one test instead of 2 then

Reply:
Thought about it, I'm not really sure we should make such assumptions. Internally Thread could save those runnables unordered and call not in the order they were added

Reply:
Also I believe it's a good practice to split checks and keep verifications as atomic as possible, tests are easier to support then

Reply:
it's true when using well-written parametrized tests or smth similar
in this case tests are not greatly refactored and imho leaving one test verifying execution order of runnables + correct invocation time between core draw calls and swapping buffers is more reasonable

Reply:
Discussed, refactored to single test